Add prototype asynchronous data block processing backend

BenjaminCharmes opened the pull request:
Closes #1469

Linked to [Add prefers_async flag to insitu blocks #76](https://github.com/datalab-org/datalab-app-plugin-insitu/pull/76)

## Summary
Implements asynchronous data block processing using APScheduler for computationally intensive blocks, with a unified task management system for all async operations.

## Changes

### Backend

**Task Management Refactoring:**
* **Unified Task model** replacing separate `BlockTask` and `ExportTask` models
* **Single `tasks` MongoDB collection** with `type` field to distinguish export vs block processing tasks
* **Consolidated status enums** - `TaskStatus` replaces `ExportStatus` and `BlockProcessingStatus`

**Async Block Processing:**
* Added Task model to track async processing status (PENDING/PROCESSING/READY/ERROR)
* Added TaskStage model to log timestamped processing stages with severity levels (info/warning/error)
* Modified `update_block()` to support async processing via `prefers_async` attribute
* Added `/blocks/<task_id>/status` endpoint for status polling, including stage history
* Reused existing export_scheduler infrastructure for background job execution
* Added AsyncCanaryBlock for testing async processing workflow

### Frontend
* Added `pollBlockStatus()` to poll task status every 2 seconds
* Modified `updateBlockFromServer()` to handle async responses
* Added real-time progress display showing current processing stage in UI
* Updated insitu block components (NMR & UV-Vis) to trigger async processing only when all required data is present

## Implementation Details
* Blocks annotate with `prefers_async = True` to enable async processing
* UI sends `trigger_async: true` in `event_data` when ready to process
* Processing stages logged at each step (starting, loading, processing, generating viz, saving, completion)
* `bokeh_plot_data` is regenerated on-demand via `block.to_web()` in status endpoint
* Only creates async tasks when all folders/parameters are selected
* Stage information displayed as blue info alert during processing, auto-cleared on completion
* Task type filtering ensures export and block processing tasks remain isolated despite sharing infrastructure

EDIT by @ml-evs:

Took over this PR and tweaked it so that block processing does not go via APScheduler, but instead uses a separate task processor per flask/gunicorn worker  that is in charge of updating the mongodb tasks collection. The results are also pushed to gridfs to use as a transfer buffer.

BenjaminCharmes committed:
Add asynchronous data block processing with APScheduler

Add async to XRDInsituBlock.vue

ml-evs reviewed (commented):
Excited to try this @BenjaminCharmes! Let me know when you think its ready to play around with! I would suggest also extending the `CanaryBlock` to make an async version we can use in testing.

BenjaminCharmes committed:
Add block processing 'stages' and display it in the UI

ml-evs commented on pydatalab/src/pydatalab/models/block_task.py:
```diff
@@ -0,0 +1,43 @@
+from datetime import datetime, timezone
```
Could we combine the block tasks and export task models and MongoDB collection?

BenjaminCharmes committed:
Unify BlockTask and ExportTask into single Task model

Fix linting

Fix Task.creator_id type to PyObjectId for validation compatibility

Add newline at end of schema files for pre-commit

Regenerate schemas with proper formatting

Regenerate schemas with proper formatting

BenjaminCharmes committed:
Refactor Task model to use spec-based architecture for type-specific fields

BenjaminCharmes committed:
Fix async block processing: authentication context and database persistence

BenjaminCharmes committed:
Manual fix pre-commit

ml-evs committed:
Enable async in echem block

ml-evs committed:
Rename export scheduler to task scheduler

ml-evs committed:
Turn on async canary block

ml-evs committed:
Re(?)introduce store setter for block updating status

ml-evs committed:
Remove unecessary file operation logging

- Add type ignore

ml-evs committed:
Allow more detail in task stages and temporarily enable async everywhere

ml-evs committed:
Allow tasks to store more detail on error

ml-evs committed:
Refactor async task processing around a gridfs-based transfer buffer

ml-evs committed:
Add log lines to async tasks

ml-evs committed:
Add scheduled task to clear up task results transfer buffer

ml-evs committed:
Tweak APScheduler settings to work better with multiple jobs

ml-evs committed:
Show timestamped processing stages in collapsible view for async blocks

ml-evs committed:
Lock async block processing behind config flag per block type

ml-evs committed:
Add test stubs for async blocks

ml-evs committed:
Remove scheduler for blocks in favour of simple processing queue

ml-evs committed:
Add docs note about async block processing

ml-evs committed:
Bump navani and in situ plugin

ml-evs committed:
Catch the case that not block data is returned after async processing

ml-evs committed:
Use dynamic backoff time when polling block status
